File: Subin/문자열 나누기.py
#정답 문제여서 코드만
def solution(s):
    answer = 0
    #x 수 세고 y 수 세고 같아질때마다 
    #x,y 초기화하고/ x값 바꾸고 결과값 +1
    s = s + ' '
    x = s[0]
    cx = 1
    cy =0
    for  i in range(1,len(s)-2):
        #숫자 세기
        if s[i] == x:
            cx +=1
            if cx == cy:
                answer +=1
                cx =0
                cy=0
                if s[i+1]==' ':
                    break
                else:
                    x= s[i+1]
        else:
            cy +=1
            if cx == cy:
                answer +=1
                cx =0
                cy=0
                if s[i+1]==' ':
                    break
                else:
                    x= s[i+1] 

    return answer +1

File: nahyun/Re3/140108. 문자열 나누기/문자열 나누기.java
class Solution {
    public int solution(String s) {
        char[] cs = new char[s.length()];
        for(int i = 0; i<s.length(); i++) {
            cs[i] = s.charAt(i);
        }
        int answer = 0;
        int xCount = 1;
        int yCount = 0; 
        char x = cs[0];
        if(s.length() == 1) return 1;
        for(int i = 1; i<s.length(); i++) {
            if(x == cs[i]) xCount++;
            else yCount++;
            if(xCount == yCount) {
                answer++;
                System.out.println(cs[i]);
                xCount = 1; yCount = 0;
                if(i+1 < s.length()) {
                    if(i+1 == s.length()-1) {
                        xCount++; break;
                    }
                    x = cs[++i];
                }
                else break;
            }
        }
        if(xCount != 1 || yCount != 0) return ++answer;
        return answer;
    }
}
